src/js/utils/PodUtil.js

```diff
@@ -0,0 +1,42 @@
+var PodUtil = {
+
+  /**
+   * Returns true if the container given matches
+   * the free-text given.
+   *
+   * @param {PodContainer} container - The Container to test
+   * @param {string} text - The text to test against
+   * @returns {boolean} - Returns true if the test passes
+   */
+  isContainerMatchingText(container, text) {
+    if (!text) {
+      return true;
+    }
+
+    return container.getName().indexOf(text) >= 0;
+  },
+
+  /**
+   * Returns true if the container instance matches
+   * the free-text given.
+   *
+   * @param {PodInstance} instance - The Container to test
+   * @param {string} text - The text to test against
+   * @returns {boolean} - Returns true if the test passes
+   */
+  isInstanceOrChildrenMatchingText(instance, text) {
```

Comment:
I am adding this on the todo because it's going to add some merging overhead on the linked PRs.